source/material_model/rheology/diffusion_dislocation.cc

```diff
@@ -88,99 +89,90 @@ namespace aspect
                                 edot_ii/prefactor_stress_diffusion
                                 :
                                 0.5 / maximum_viscosity);
-            double log_stress_ii = std::log(stress_ii);
-            double log_strain_rate_residual = 2 * log_strain_rate_residual_threshold;
-            double log_strain_rate_deriv = 0;
-            unsigned int stress_iteration = 0;
-            while (std::abs(log_strain_rate_residual) > log_strain_rate_residual_threshold
-                   && stress_iteration < stress_max_iteration_number)
```

Comment:
This is your current stopping criterion. Could you check how many iterations you are using in this loop (in total or on average) in one of your benchmarks, and compare that against the number of iterations you're spending in KINSOLS? I think that might clarify why the code is so much slower.

You can also directly time individual sections by querying `simulator_access.get_computing_timer()`. You can take what's in `particles/world.cc` as an example of how you can put a timing section around a piece of code.